src/openzeppelin/access/accesscontrol.cairo

```diff
@@ -0,0 +1,174 @@
+# SPDX-License-Identifier: MIT
+# OpenZeppelin Cairo Contracts v0.x.0 (access/accesscontrol.cairo)
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.starknet.common.syscalls import get_caller_address
+
+from openzeppelin.introspection.ERC165 import ERC165_register_interface
+from openzeppelin.utils.constants import TRUE, FALSE, IACCESSCONTROL_ID
+
+#
+# Events
+#
+
+@event
+func RoleGranted(role: felt, account: felt, sender: felt):
+end
+
+@event
+func RoleRevoked(role: felt, account: felt, sender: felt):
+end
+
+@event
+func RoleAdminChanged(role: felt, previousAdminRole: felt, newAdminRole: felt):
+end
+
+#
+# Storage
+#
+
+@storage_var
+func _roleAdmin(role: felt) -> (admin: felt):
+end
+
+@storage_var
+func _roleMember(role: felt, account: felt) -> (hasRole: felt):
+end
+
+#
+# Constructor
+#
+
+func AccessControl_initializer{
+    syscall_ptr : felt*,
+    pedersen_ptr : HashBuiltin*,
+    range_check_ptr
+}():
+    ERC165_register_interface(IACCESSCONTROL_ID)
+    return ()
+end
+
+#
+# Modifier
+#
+
+func AccessControl_onlyRole{
+    syscall_ptr : felt*,
+    pedersen_ptr : HashBuiltin*,
+    range_check_ptr
+}(role: felt):
+    let (caller: felt) = get_caller_address()
+    let (hasRole: felt) = AccessControl_hasRole(role, caller)
+    with_attr error_message("AccessControl: caller is missing role"):
```

Comment:
Could you add the `role` value to the error string ([ref](https://www.cairo-lang.org/docs/how_cairo_works/scope_attributes.html)). I know it will just be a number but at least that's easily convertible to a string and can help with debugging.